app/client/components/consent/PrivacySettings.tsx

```diff
@@ -537,3 +501,40 @@ export class PrivacySettings extends Component<{}, State> {
     }));
   }
 }
+
+const close = () => {
+  window.parent.postMessage("closeCmp", "*");
+};
+
+const doScrolling = (query: string, duration: number): void => {
+  const element: HTMLElement | null = document.querySelector(query);
+  if (!element) {
+    return;
+  }
+
+  const elementY: number =
+    window.pageYOffset + element.getBoundingClientRect().top;
+  const startingY: number = window.pageYOffset;
+  const diff: number = elementY - startingY;
+  let start: number;
+
+  const animationStep: FrameRequestCallback = timestamp => {
+    if (!start) {
+      start = timestamp;
+    }
+    // Elapsed milliseconds since start of scrolling.
+    const time: number = timestamp - start;
+    // Get percent of completion in range [0, 1].
+    const percent: number = Math.min(time / duration, 1);
+
+    window.scrollTo(0, startingY + diff * percent);
+
+    // Proceed with animation as long as we wanted it to.
+    if (time < duration) {
+      window.requestAnimationFrame(animationStep);
+    }
+  };
+
+  // Bootstrap our animation - it will get called right before next frame shall be rendered.
+  window.requestAnimationFrame(animationStep);
```

Comment:
how well does this play with React?

Reply:
We've tested it and it seems to behave as expected